modules/gateway-task/main.tf

```diff
@@ -53,6 +53,61 @@ locals {
 
   https_ca_cert_arn = var.consul_https_ca_cert_arn != "" ? var.consul_https_ca_cert_arn : var.consul_ca_cert_arn
   grpc_ca_cert_arn  = var.consul_grpc_ca_cert_arn != "" ? var.consul_grpc_ca_cert_arn : var.consul_ca_cert_arn
+
+  mesh_init_container_definition = {
+    name             = "consul-ecs-mesh-init"
+    image            = var.consul_ecs_image
+    essential        = false
+    logConfiguration = var.log_configuration
+    command          = ["mesh-init"]
+    mountPoints = [
+      local.consul_data_mount_read_write,
+      {
+        sourceVolume  = local.consul_binary_volume_name
+        containerPath = "/bin/consul-inject"
+        readOnly      = true
+      }
+    ]
+    cpu         = 0
+    volumesFrom = []
+    environment = [
+      {
+        name  = "CONSUL_ECS_CONFIG_JSON",
+        value = local.encoded_config
+      }
+    ]
+    linuxParameters = {
+      initProcessEnabled = true
+      capabilities       = var.enable_transparent_proxy ? { add = ["NET_ADMIN"] } : {}
+    }
+    secrets = flatten(
+      concat(
+        var.tls ? [
+          concat(
+            local.https_ca_cert_arn != "" ? [
+              {
+                name      = "CONSUL_HTTPS_CACERT_PEM"
+                valueFrom = local.https_ca_cert_arn
+              },
+            ] : [],
+            local.grpc_ca_cert_arn != "" ? [
+              {
+                name      = "CONSUL_GRPC_CACERT_PEM"
+                valueFrom = local.grpc_ca_cert_arn
+              }
+            ] : [],
+            []
+          )
+        ] : [],
+        []
+      )
+    )
+  }
+
+  # Additional user attribute that needs to be added to run the mesh-init
+  # container with root access.
+  additional_user_attr                     = var.enable_transparent_proxy ? { user = "0" } : {}
```

Comment:
Question - When `var.enable_transparent_proxy` is false what will be the value of `user` in `finalized_mesh_init_container_definition` ?

Reply:
I think it will be empty or the user who starts the container. Definitely not `root`